docs/contributing/crd-source.md

```diff
@@ -0,0 +1,104 @@
+# CRD Source
+
+CRD source provides a generic mechanism to manage DNS records in your favourite DNS provider supported by external-dns.
+
+### Details
+
+CRD source watches for the user specified CRD to extract [Endpoints](https://github.com/kubernetes-incubator/external-dns/blob/master/endpoint/endpoint.go) from its `Spec`.
+So users need to create such a CRD and register it to the kubernetes cluster and then create new object(s) of the CRD specifying the Endpoints.
+
+### Registering CRD
+
+Here is typical example of CRD which provides Endpoints to `CRD source`:
```

Comment:
You should link to the example here so people know that there's an example the rest of this file is in terms of.

Reply:
Added.